experiments/experiment3_3.sh

```diff
@@ -32,45 +32,45 @@ echo "initializing..."
+    python ../utils/dataset_creator.py "binarize" "$csv_base_path/$city/$train_set_csv" "$label" "$csv_base_path/tmp/$city/train_set$label.csv"
+    python ../utils/dataset_creator.py "binarize" "$csv_base_path/$city/$test_set_csv" "$label" "$csv_base_path/tmp/$city/test_set$label.csv"
   done
 done
 
 for subsetted_city in ${cities[@]}; do
   for subset in ${subsets[@]}; do
     # make relevant directories
-    mkdir -p $csv_base_path/"tmp/subset_"$subsetted_city/$subset/
-    mkdir -p $model_save_folder/$experiment/"subset_"$subsetted_city/$subset/
-    mkdir -p $visualizations_path/$experiment/$subsetted_city/$subset/
+    mkdir -p "$csv_base_path/tmp/subset_$subsetted_city/$subset/"
+    mkdir -p "$model_save_folder/$experiment/subset_$subsetted_city/$subset/"
+    mkdir -p "$visualizations_path/$experiment/$subsetted_city/$subset/"
     
     for label in {1..4}; do
-      echo "training label "$label" classifier on all cities with a "$subset" subset of "$subsetted_city"..."
+      echo "training label $label classifier on all cities with a $subset subset of $subsetted_city..."
       # combine subset of city of interest with full train sets of all other cities
       arguments=""
       for city in ${cities[@]}; do
         if [ $city == $subsetted_city ]; then
-          dataset_size=$(($(wc $csv_base_path/"tmp/"$city/"train_set"$label".csv" | awk '{print $1}') - 1))
+          dataset_size=$(($(wc $csv_base_path/tmp/$city/train_set$label.csv | awk '{print $1}') - 1))
           subset_size=$(echo $subset*$dataset_size | bc)
-          python ../utils/dataset_creator.py "subset" $csv_base_path/"tmp/"$city/"train_set"$label".csv" $subset_size $csv_base_path/"tmp/subset.csv"
+          python ../utils/dataset_creator.py "subset" "$csv_base_path/tmp/$city/train_set$label.csv" "$subset_size" "$csv_base_path/tmp/subset.csv"
           arguments+="$csv_base_path/tmp/subset.csv "
         else
           arguments+="$csv_base_path/tmp/$city/train_set$label.csv "
         fi
       done
-      python ../utils/dataset_creator.py "combine" $arguments $csv_base_path/"tmp/subset_"$subsetted_city/$subset/"train_set"$label".csv"
+      python ../utils/dataset_creator.py "combine" "$arguments" "$csv_base_path/tmp/subset_$subsetted_city/$subset/train_set$label.csv"
       rm "$csv_base_path/tmp/subset.csv"
 
       # train model on combined train set
-      python ../train.py $model_name$label $image_base_path/ $csv_base_path/"tmp/subset_"$subsetted_city/$subset/"train_set"$label".csv" $model_name $model_save_folder/$experiment/"subset_"$subsetted_city/$subset/ $num_epochs $crop_size
+      python ../train.py "$model_name$label" "$image_base_path/" "$csv_base_path/tmp/subset_$subsetted_city/$subset/train_set$label.csv" "$model_name" "$model_save_folder/$experiment/subset_$subsetted_city/$subset/" "$num_epochs" "$crop_size"
```

Comment:
Can we use the same naming convention as in the other experiment scripts for the saved model
`"${experiment}_${model_name}_$label"`